app/lib/payment_gateways/stripe_crypt.rb

```diff
@@ -44,8 +46,12 @@ def find_or_create_customer
       customer_id = payment_detail.credit_card_auth_code
       return create_customer if customer_id.blank?
 
-      customer = Stripe::Customer.retrieve(customer_id, api_key)
-      customer.deleted? ? create_customer : customer
+      begin
+        customer = Stripe::Customer.retrieve(customer_id, api_key)
+        customer.deleted? ? create_customer : customer
+      rescue Stripe::InvalidRequestError => e
+        puts "An invalid request occurred."
```

Comment:
We shouldn't use `puts` in the code. In this case `Rails.logger.warn` could be used.